app/main.py

```diff
@@ -1,2 +1,75 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    key_hash: int
+
+
 class Dictionary:
     pass
+    INITIAL_CAPACITY = 8
+    THRESHOLD = 2 / 3
+    CAPACITY_MULTIPLIER = 2
+
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self._size = 0
+        self._hash_table: list[None | Node] = [None] * self.capacity
+
+    def __repr__(self) -> str:
+        return str(self)
```

Comment:
The `__repr__` method calls `str(self)`, which will lead to infinite recursion because `str(self)` calls `__str__`, which calls `str(self._hash_table)`. Consider returning a more informative string representation of the object, such as `f"Dictionary({self._hash_table})"`.